catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -314,17 +278,89 @@ protected Object executeWithSubject() throws Exception {
         return null;
     }
 
+    private File getInputFile() {
```

Comment:
As it's invoked in only one place locally and immediately tested, I'm not sure there's a lot of difference either way. It just requires the creation of Optionals with nulls in them when the input file doesn't exist or should be skipped.